teps/0059-skip-guarded-task-only.md

```diff
@@ -0,0 +1,225 @@
+
+```
+          lint                     unit-tests
+           |                           |
+           v                           v
+   report-linter-output        integration-tests
+                                       |
+                                       v
+                                 manual-approval
+                                       |
+                                       v
+                                  build-image
+                                       |
+                                       v
+                                  deploy-image
+```
+
+Today, if `manual-approval` is skipped then `build-image` and `deploy-image` would be skipped as well while `lint` and `report-linter-output` would execute. In this TEP, we'll provide the flexibility to execute `build-image` and `deploy-image` when `manual-approval` is skipped. This would allow the user to reuse the `Pipeline` in both scenarios.
+
+Building on the above use case, the user adds `slack-msg` which sends a notification to slack that it was manually approved with the name of the approver that is passed as a `Result` from `manual-approval` to `slack-msg`.
+
+```
+          lint                     unit-tests
+           |                           |
+           v                           v
+   report-linter-output        integration-tests
+                                       |
+                                       v
+                                 manual-approval
+                                 |            |
+                                 v        (approver)
+                            build-image       |
+                                |             v
+                                v          slack-msg
+                            deploy-image
+```
+
+If the guarded `manual-approval` is skipped, then `build-image` and `deploy-image` needs to be executed similarly to above. However, `slack-msg` should be skipped because of the missing `Result` reference to the approver name.
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accomodated.
+-->
+
+- Users should be able to specify that a guarded `Task` and its dependent `Tasks` can be should be skipped when its `WhenExpressions` evaluate to `False`
+- Users should be able to specify that a guarded `Task` only should be skipped when its `WhenExpressions` evaluate to `False`, and unblock the execution of its dependent `Tasks`
+  - its ordering-dependent `Tasks`, based on `runAfter`, should execute as expected
+  - its resource-dependent `Tasks`, based on resources such as `Results`, should be attempted but skipped because they can't resolve missing resources
```

Comment:
`should be attempted but skipped because they can't resolve ...`: the wording is a bit unclear to me, I think this should say `should be attempted but might be skipped if they can't resolve ...`